### ISSUE TITLE
Operator invocations don't require `unsafe` context when pointers are involved

### ISSUE THREAD

**Issue by jjonescz** [dotnet/roslyn#81967]
**Version Used**: [5.3.0-2.25619.2](https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-tools/NuGet/Microsoft.Net.Compilers.Toolset/overview/5.3.0-2.25619.2) ([6a135ea](https://github.com/dotnet/roslyn/commit/6a135ea0ca8b7332d11fa78d68f0a14a65f2644f))

**Steps to Reproduce**:

```cs
var c = new C();
c -= null; // no error
c = c + null; // no error
c.M(null); // error

public class C
{
    public unsafe void operator -=(int* p) { }
    public static unsafe C operator +(C c1, int* p) => c1;
    public unsafe void M(int* p) { }
}
```

**Expected Behavior**: The operator invocations should report "error CS0214: Pointers and fixed size buffers may only be used in an unsafe context" just like the method invocation does.

**Actual Behavior**: No errors for the operator invocations.

Fixing this would be a breaking change.
